File: model/Categoria.js

const { Sequelize, DataTypes } = require('sequelize');

const bd = require('./bd')
const sequelize = bd.sequelize

const Categoria = sequelize.define('categoria', {
	nombre: {
		type: DataTypes.STRING(30),
		allowNull: false,
		unique: true
	},
	activo: {
		type: DataTypes.BOOLEAN,
		defaultValue: true
	},
	imagen: {
		type: DataTypes.STRING(1000)
	},
	ruta: {
		type: DataTypes.STRING(300)
	},
}, {
	sequelize,
	timestamps: true,
	createdAt: 'f_creacion',
	updatedAt: 'f_edicion',
    modelName: 'Categoria',
    //freezeTableName: true, // Nombre de la tabla igual al del modelo
    tableName: 'categorias',
    underscored: true,// Permitir "_" en campos/columnas FOREIGN KEY ???
})

const init = async () => {
	await sequelize.sync();
	console.log('TB Categoria ready ...!');
}
init()

module.exports = Categoria;

File: front/categorias/domain/entities/categoria.dart
class Categoria {
  Categoria({
    this.id,
    this.nombre,
    this.activo,
    this.imagen,
    this.ruta,
    this.fCreacion,
    this.fEdicion,
  });

  final int id;
  final String nombre;
  final bool activo;
  final String imagen;
  final String ruta;
  final DateTime fCreacion;
  final DateTime fEdicion;
}
